File: src/main/java/xyz/chunkstories/api/net/packets/PacketText.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.net.packets;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

import xyz.chunkstories.api.net.Packet;
import xyz.chunkstories.api.net.PacketDestinator;
import xyz.chunkstories.api.net.PacketSender;
import xyz.chunkstories.api.net.PacketSendingContext;
import xyz.chunkstories.api.net.PacketReceptionContext;

public class PacketText extends Packet {
	public String text;

	public PacketText() {

	}

	public PacketText(String text) {
		this.text = text;
	}

	@Override
	public void send(PacketDestinator destinator, DataOutputStream out, PacketSendingContext context) throws IOException {
		out.writeUTF(text);
	}

	public void process(PacketSender sender, DataInputStream in, PacketReceptionContext context) throws IOException {
		text = in.readUTF();

		// Actual handling is left to internal code defined in proprietary subclasses
		// that expose the inner workings
		// ( see clientClass and serverClass within .packets files )
	}
}

File: src/main/java/xyz/chunkstories/api/net/packets/PacketText.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.net.packets

import xyz.chunkstories.api.Engine
import xyz.chunkstories.api.net.Packet
import xyz.chunkstories.api.server.UserConnection
import java.io.DataInputStream
import java.io.DataOutputStream

open class PacketText(engine: Engine) : Packet(engine) {
    lateinit var text: String

    constructor(engine: Engine, text: String) : this(engine) {
        this.text = text
    }

    override fun send(dos: DataOutputStream) {
        dos.writeUTF(text)
    }

    override fun receive(dis: DataInputStream, user: UserConnection?) {
        text = dis.readUTF()
    }
}